tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ConstSimplifierBase.scala

```diff
@@ -113,130 +128,93 @@ abstract class ConstSimplifierBase {
-      ValEx(TlaBool(true))(boolTag)
-
+    // TRUE -> x = x
+    case OperEx(TlaBoolOper.implies, ValEx(TlaBool(true)), right) => right
+    // x -> FALSE = !x
     case OperEx(TlaBoolOper.implies, lhs, ValEx(TlaBool(false))) =>
       simplifyShallow(OperEx(TlaBoolOper.not, lhs)(boolTag))
 
+    // Evaluate equivalence between constants
     case OperEx(TlaBoolOper.equiv, ValEx(TlaBool(left)), ValEx(TlaBool(right))) =>
       ValEx(TlaBool(left == right))(boolTag)
 
-    case OperEx(TlaBoolOper.equiv, ValEx(TlaBool(left)), right) =>
-      if (left) {
-        right
-      } else {
-        simplifyShallow(OperEx(TlaBoolOper.not, right)(boolTag))
-      }
-
-    case OperEx(TlaBoolOper.equiv, left, ValEx(TlaBool(right))) =>
-      if (right) {
-        left
-      } else {
-        simplifyShallow(OperEx(TlaBoolOper.not, left)(boolTag))
-      }
+    // TRUE <=> x = x
+    case OperEx(TlaBoolOper.equiv, ValEx(TlaBool(true)), right) => right
+    // FALSE <=> x = !x
+    case OperEx(TlaBoolOper.equiv, ValEx(TlaBool(false)), right) =>
+      simplifyShallow(OperEx(TlaBoolOper.not, right)(boolTag))
+    // x <=> TRUE = x
+    case OperEx(TlaBoolOper.equiv, left, ValEx(TlaBool(true))) => left
+    // x <=> FALSE = !x
+    case OperEx(TlaBoolOper.equiv, left, ValEx(TlaBool(false))) =>
+      simplifyShallow(OperEx(TlaBoolOper.not, left)(boolTag))
 
     // many ite expressions can be simplified like this
-    case OperEx(TlaControlOper.ifThenElse, ValEx(TlaBool(true)), thenEx, _) =>
-      thenEx
-
-    case OperEx(TlaControlOper.ifThenElse, ValEx(TlaBool(false)), _, elseEx) =>
-      elseEx
-
-    case OperEx(TlaControlOper.ifThenElse, pred, ValEx(TlaBool(false)), elseEx) =>
-      OperEx(TlaBoolOper.and, OperEx(TlaBoolOper.not, pred)(boolTag), elseEx)(boolTag)
-
-    case OperEx(TlaControlOper.ifThenElse, pred, ValEx(TlaBool(true)), ValEx(TlaBool(false))) =>
-      pred
-
+    // IF true THEN x ELSE y = x
+    case OperEx(TlaControlOper.ifThenElse, ValEx(TlaBool(true)), thenEx, _) => thenEx
+    // IF false THEN x ELSE y = y
+    case OperEx(TlaControlOper.ifThenElse, ValEx(TlaBool(false)), _, elseEx) => elseEx
+    // IF x THEN TRUE ELSE FALSE = x
+    case OperEx(TlaControlOper.ifThenElse, pred, ValEx(TlaBool(true)), ValEx(TlaBool(false))) => pred
+    // IF x THEN FALSE ELSE TRUE = !x
     case OperEx(TlaControlOper.ifThenElse, pred, ValEx(TlaBool(false)), ValEx(TlaBool(true))) =>
-      OperEx(TlaBoolOper.not, pred)(boolTag)
-
-    case ite @ OperEx(TlaControlOper.ifThenElse, _, thenEx, elseEx) =>
-      if (thenEx != elseEx) {
-        ite
-      } else {
-        thenEx
-      }
+      simplifyShallow(OperEx(TlaBoolOper.not, pred)(boolTag))
+    // IF x THEN FALSE ELSE y = !x /\ y
+    case OperEx(TlaControlOper.ifThenElse, pred, ValEx(TlaBool(false)), elseEx) =>
+      val newPredicate = simplifyShallow(OperEx(TlaBoolOper.not, pred)(boolTag))
+      simplifyShallow(OperEx(TlaBoolOper.and, newPredicate, elseEx)(boolTag))
```

Comment:
Just for symmetry, do we also want `IF x THEN TRUE ELSE y <=> x \/ y` ? What about the `IF x then y ELSE (T/F)` cases?